examples/ecies_example.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <mococrw/ecies.h>
+#include <mococrw/key.h>
+
+#include <boost/program_options.hpp>
+
+using namespace mococrw;
+namespace po = boost::program_options;
+
+struct EciesData{
+    std::shared_ptr<AsymmetricPublicKey> ephKey;
+    std::shared_ptr<AsymmetricPrivateKey> privKey;
+    std::shared_ptr<AsymmetricPublicKey> pubKey;
+    std::shared_ptr<X509Certificate> cert;
+    std::shared_ptr<ECCSpec> eccSpec;
```

Comment:
All these objects are lightweight wrappers around a `std::shared_ptr<>` as far as I know. As a consequence, there's little use of wrapping them in another shared pointer, you can just copy them by value instead. That will also simplify some of the code you use above, like `encBuilder.buildEncryptionCtx(*eciesData.pubKey.get());`